cli/cli-client/pom.xml

```diff
@@ -118,12 +123,134 @@
                      <goal>java</goal>
                   </goals>
                   <configuration>
-                     <mainClass>org.infinispan.cli.CLI</mainClass>
+                     <mainClass>${cli.main.class}</mainClass>
                      <cleanupDaemonThreads>false</cleanupDaemonThreads>
                   </configuration>
                </execution>
             </executions>
          </plugin>
       </plugins>
    </build>
+
+   <profiles>
+      <profile>
+         <id>shade</id>
+         <build>
+            <plugins>
+               <plugin>
+                  <groupId>org.apache.maven.plugins</groupId>
+                  <artifactId>maven-shade-plugin</artifactId>
+                  <executions>
+                     <execution>
+                        <phase>package</phase>
+                        <goals>
+                           <goal>shade</goal>
+                        </goals>
+                        <configuration>
+                           <transformers>
+                              <transformer implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
+                                 <mainClass>${cli.main.class}</mainClass>
+                              </transformer>
+                           </transformers>
+                        </configuration>
+                     </execution>
+                  </executions>
+               </plugin>
+            </plugins>
+         </build>
+      </profile>
+      <profile>
+         <id>native</id>
+         <build>
+            <plugins>
+               <plugin>
+                  <groupId>org.apache.maven.plugins</groupId>
+                  <artifactId>maven-enforcer-plugin</artifactId>
+                  <executions>
+                     <execution>
+                        <id>graalvm-check</id>
+                        <goals>
+                           <goal>enforce</goal>
+                        </goals>
+                        <configuration>
+                           <rules>
+                              <requireEnvironmentVariable>
+                                 <variableName>GRAALVM_HOME</variableName>
+                              </requireEnvironmentVariable>
+                           </rules>
+                           <fail>true</fail>
+                        </configuration>
+                     </execution>
+                  </executions>
+               </plugin>
+               <plugin>
```

Comment:
is this plugin required? I don't see any reference to the `lib` folder.

Reply:
It puts the classpath in a property which is then used by the native image build